transport/socks5/stream_dialer.go

```diff
@@ -30,23 +30,26 @@ type credentials struct {
 	password []byte
 }
 
-// NewStreamDialer creates a [transport.StreamDialer] that routes connections to a SOCKS5
+// NewDialer creates a [transport.StreamDialer] that routes connections to a SOCKS5
 // proxy listening at the given [transport.StreamEndpoint].
-func NewStreamDialer(endpoint transport.StreamEndpoint) (*StreamDialer, error) {
-	if endpoint == nil {
+func NewClient(streamEndpoint transport.StreamEndpoint) (*Client, error) {
+	if streamEndpoint == nil {
 		return nil, errors.New("argument endpoint must not be nil")
 	}
-	return &StreamDialer{proxyEndpoint: endpoint, cred: nil}, nil
+	return &Client{se: streamEndpoint, cred: nil}, nil
 }
 
-type StreamDialer struct {
-	proxyEndpoint transport.StreamEndpoint
-	cred          *credentials
+type Client struct {
+	se   transport.StreamEndpoint
+	pe   transport.PacketEndpoint
```

Comment:
Remove the PacketEndpoint, it's not needed.

Reply:
@amircybersec thanks for catching this. yes it is redundant.